### ISSUE TITLE
[glib] Build failed on installing glib:x64-osx

### ISSUE THREAD

**Issue by magicfoo** [microsoft/vcpkg#9012]
submit an issue at https://github.com/Microsoft/vcpkg/issues including:
  Package: glib:x64-osx
  Vcpkg version: 2019.09.12-unknownhash

configure: error: in `/Users/boss/p4/vcpkg/buildtrees/glib/x64-osx-dbg/config':
configure: error: cannot run C compiled programs.
If you meant to cross compile, use `--host'.
See `config.log' for more details
CMake Error at CMakeLists.txt:99 (message):
  Configure failed.

/Users/boss/p4/vcpkg/buildtrees/glib/config-x64-osx-dbg-err.log (END)

**Comment by NancyLi1013** [comment]
Hi @magicfoo thanks for posting this issue.
Could you please attach `/Users/boss/p4/vcpkg/buildtrees/glib/x64-osx-dbg/config.log`?

**Comment by Pochatkin** [comment]
I also have a problem with building glib:x64-osx.

    The following packages will be built and installed:
        glib[core]:x64-osx
    Starting package 1/1: glib:x64-osx
    Building package glib[core]:x64-osx...
    A suitable version of ninja was not found (required v1.8.2). Downloading portable ninja v1.8.2...
    Downloading ninja...
      https://github.com/ninja-build/ninja/releases/download/v1.8.2/ninja-mac.zip -> /Users/jetbrains/dev/vcpkg/downloads/ninja-mac-1.8.2.zip
      % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                     Dload  Upload   Total   Spent    Left  Speed
    100   602    0   602    0     0    882      0 --:--:-- --:--:-- --:--:--   881
    100 77284  100 77284    0     0  52681      0  0:00:01  0:00:01 --:--:--  187k
    Extracting ninja...
    -- Downloading https://ftp.gnome.org/pub/gnome/sources/glib/2.52/glib-2.52.3.tar.xz...
    -- Extracting source /Users/jetbrains/dev/vcpkg/downloads/glib-2.52.3.tar.xz
    -- Applying patch use-libiconv-on-windows.patch
    -- Applying patch arm64-defines.patch
    -- Applying patch fix-arm-builds.patch
    -- Using source at /Users/jetbrains/dev/vcpkg/buildtrees/glib/src/2.52.3-34a15219ec
    -- Configuring x64-osx-dbg
    CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:72 (message):
        Command failed: /usr/local/Cellar/cmake/3.15.5/bin/cmake /Users/jetbrains/dev/vcpkg/buildtrees/glib/src/2.52.3-34a15219ec -DHAVE_SELINUX=OFF -DGLIB_VERSION=2.52.3 -DCMAKE_MAKE_PROGRAM=/Users/jetbrains/dev/vcpkg/downloads/tools/ninja-1.8.2-osx/ninja -DCMAKE_SYSTEM_NAME=Darwin -DBUILD_SHARED_LIBS=OFF -DVCPKG_CHAINLOAD_TOOLCHAIN_FILE=/Users/jetbrains/dev/vcpkg/scripts/toolchains/osx.cmake -DVCPKG_TARGET_TRIPLET=x64-osx -DVCPKG_SET_CHARSET_FLAG=ON -DVCPKG_PLATFORM_TOOLSET=external -DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON -DCMAKE_FIND_PACKAGE_NO_PACKAGE_REGISTRY=ON -DCMAKE_FIND_PACKAGE_NO_SYSTEM_PACKAGE_REGISTRY=ON -DCMAKE_INSTALL_SYSTEM_RUNTIME_LIBS_SKIP=TRUE -DCMAKE_VERBOSE_MAKEFILE=ON -DVCPKG_APPLOCAL_DEPS=OFF -DCMAKE_TOOLCHAIN_FILE=/Users/jetbrains/dev/vcpkg/scripts/buildsystems/vcpkg.cmake -DCMAKE_ERROR_ON_ABSOLUTE_INSTALL_DESTINATION=ON -DVCPKG_CXX_FLAGS= -DVCPKG_CXX_FLAGS_RELEASE= -DVCPKG_CXX_FLAGS_DEBUG= -DVCPKG_C_FLAGS= -DVCPKG_C_FLAGS_RELEASE= -DVCPKG_C_FLAGS_DEBUG= -DVCPKG_CRT_LINKAGE=dynamic -DVCPKG_LINKER_FLAGS= -DVCPKG_TARGET_ARCHITECTURE=x64 -DCMAKE_INSTALL_LIBDIR:STRING=lib -DCMAKE_INSTALL_BINDIR:STRING=bin -DGLIB_SKIP_HEADERS=ON -DGLIB_SKIP_TOOLS=ON -G Ninja -DCMAKE_BUILD_TYPE=Debug -DCMAKE_INSTALL_PREFIX=/Users/jetbrains/dev/vcpkg/packages/glib_x64-osx/debug
        Working Directory: /Users/jetbrains/dev/vcpkg/buildtrees/glib/x64-osx-dbg
        Error code: 1
        See logs for more information:
          /Users/jetbrains/dev/vcpkg/buildtrees/glib/config-x64-osx-dbg-out.log
          /Users/jetbrains/dev/vcpkg/buildtrees/glib/config-x64-osx-dbg-err.log

    Call Stack (most recent call first):
      scripts/cmake/vcpkg_configure_cmake.cmake:305 (vcpkg_execute_required_process)
      ports/glib/portfile.cmake:43 (vcpkg_configure_cmake)
      scripts/ports.cmake:94 (include)


    Error: Building package glib:x64-osx failed with: BUILD_FAILED

[config-x64-osx-dbg-err.log](https://github.com/microsoft/vcpkg/files/3851413/config-x64-osx-dbg-err.log)
[config-x64-osx-dbg-out.log](https://github.com/microsoft/vcpkg/files/3851414/config-x64-osx-dbg-out.log)
[CMakeOutput.log](https://github.com/microsoft/vcpkg/files/3851427/CMakeOutput.log)

**Comment by magicfoo** [comment]
In my specific case, I want to build cairo lib. Cairo is commonly used by GTK and I guess the glib is required to build cairo-gobject for gtk. It could be nice to have glib has an extension for cairo actually referenced by gtk only. I am not building gtk but only cairo and I don't need this dependence. I have another problem with glib on x86-windows-static (glib doesn't support static build) and removing glib/gobject from cairo works well. I will try this fix on mac.

**Comment by magicfoo** [comment]
I don't have the same logs than potchatkin. Logs are attached.

[config-x64-osx-dbg-err.log](https://github.com/microsoft/vcpkg/files/3853550/config-x64-osx-dbg-err.log)
[config-x64-osx-dbg-out.log](https://github.com/microsoft/vcpkg/files/3853551/config-x64-osx-dbg-out.log)
[config.log](https://github.com/microsoft/vcpkg/files/3853552/config.log)

**Comment by magicfoo** [comment]
problem fixed by running the following cmd line:
open /Library/Developer/CommandLineTools/Packages/macOS_SDK_headers_for_macOS_10.14.pkg

more details here: https://github.com/neovim/neovim/issues/9050#issuecomment-424417456

tracked down from: https://github.com/microsoft/vcpkg/issues/6406#issuecomment-492078829

thx all!

**Comment by magicfoo** [comment]
I reopened this issue as something as to be done to prevent this failure again.
suggested options:
- detecting the issue (missing header files in /usr/include?) and displaying a message to update the system by running the suggested cmd line
- by fixing the failing tool to look into the right location as described in section "Command Line Tools" here: https://developer.apple.com/documentation/xcode_release_notes/xcode_10_release_notes

thx

**Comment by offlinemark** [comment]
I think I had the same issue. In my case, I was trying to install `libxmlpp`.

```
Starting package 9/11: glib:x64-osx
Building package glib[core]:x64-osx...
-- Downloading https://ftp.gnome.org/pub/gnome/sources/glib/2.52/glib-2.52.3.tar.xz...
-- Extracting source /Users/mark/code/lang/cpp/vcpkg/downloads/glib-2.52.3.tar.xz
-- Applying patch use-libiconv-on-windows.patch
-- Applying patch arm64-defines.patch
-- Applying patch fix-arm-builds.patch
-- Using source at /Users/mark/code/lang/cpp/vcpkg/buildtrees/glib/src/2.52.3-34a15219ec
-- Configuring x64-osx-dbg
CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:72 (message):
    Command failed: /Users/mark/code/lang/cpp/vcpkg/downloads/tools/cmake-3.14.0-osx/cmake-3.14.0-Darwin-x86_64/CMake.app/Contents/bin/cmake /Users/mark/code/lang/cpp/vcpkg/build
trees/glib/src/2.52.3-34a15219ec -DHAVE_SELINUX=OFF -DGLIB_VERSION=2.52.3 -DCMAKE_MAKE_PROGRAM=/usr/local/bin/ninja -DCMAKE_SYSTEM_NAME=Darwin -DBUILD_SHARED_LIBS=OFF -DVCPKG_CHA
INLOAD_TOOLCHAIN_FILE=/Users/mark/code/lang/cpp/vcpkg/scripts/toolchains/osx.cmake -DVCPKG_TARGET_TRIPLET=x64-osx -DVCPKG_SET_CHARSET_FLAG=ON -DVCPKG_PLATFORM_TOOLSET=external -D
CMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON -DCMAKE_FIND_PACKAGE_NO_PACKAGE_REGISTRY=ON -DCMAKE_FIND_PACKAGE_NO_SYSTEM_PACKAGE_REGISTRY=ON -DCMAKE_INSTALL_SYSTEM_RUNTIME_LIBS_SKIP=TRUE -
DCMAKE_VERBOSE_MAKEFILE=ON -DVCPKG_APPLOCAL_DEPS=OFF -DCMAKE_TOOLCHAIN_FILE=/Users/mark/code/lang/cpp/vcpkg/scripts/buildsystems/vcpkg.cmake -DCMAKE_ERROR_ON_ABSOLUTE_INSTALL_DES
TINATION=ON -DVCPKG_CXX_FLAGS= -DVCPKG_CXX_FLAGS_RELEASE= -DVCPKG_CXX_FLAGS_DEBUG= -DVCPKG_C_FLAGS= -DVCPKG_C_FLAGS_RELEASE= -DVCPKG_C_FLAGS_DEBUG= -DVCPKG_CRT_LINKAGE=dynamic -D
VCPKG_LINKER_FLAGS= -DVCPKG_TARGET_ARCHITECTURE=x64 -DCMAKE_INSTALL_LIBDIR:STRING=lib -DCMAKE_INSTALL_BINDIR:STRING=bin -DGLIB_SKIP_HEADERS=ON -DGLIB_SKIP_TOOLS=ON -G Ninja -DCMA
KE_BUILD_TYPE=Debug -DCMAKE_INSTALL_PREFIX=/Users/mark/code/lang/cpp/vcpkg/packages/glib_x64-osx/debug
    Working Directory: /Users/mark/code/lang/cpp/vcpkg/buildtrees/glib/x64-osx-dbg
    Error code: 1
    See logs for more information:
      /Users/mark/code/lang/cpp/vcpkg/buildtrees/glib/config-x64-osx-dbg-out.log
      /Users/mark/code/lang/cpp/vcpkg/buildtrees/glib/config-x64-osx-dbg-err.log

Call Stack (most recent call first):
  scripts/cmake/vcpkg_configure_cmake.cmake:302 (vcpkg_execute_required_process)
  ports/glib/portfile.cmake:43 (vcpkg_configure_cmake)
  scripts/ports.cmake:94 (include)


Error: Building package glib:x64-osx failed with: BUILD_FAILED
Please ensure you're using the latest portfiles with `.\vcpkg update`, then
submit an issue at https://github.com/Microsoft/vcpkg/issues including:
  Package: glib:x64-osx
  Vcpkg version: 2019.09.12-unknownhash

Additionally, attach any relevant sections from the log files above.
```

[config-x64-osx-dbg-err.log](https://github.com/microsoft/vcpkg/files/3935900/config-x64-osx-dbg-err.log)
[config-x64-osx-dbg-out.log](https://github.com/microsoft/vcpkg/files/3935901/config-x64-osx-dbg-out.log)
[CMakeOutput.log](https://github.com/microsoft/vcpkg/files/3935902/CMakeOutput.log)

**Comment by gourdi** [comment]
Hello there. What's is the status of this issue? I am on 10.15, and the package macOS_SDK_headers_for_macOS_10.14.pkg is not shipped anymore, so I cannot build glib (as a dependency of Pango actually).

Any known workarounds?

Thx!

**Comment by NancyLi1013** [comment]
@gourdi 

You might try to do this and rebuild it.

`set(ENV{SDKROOT} ${CMAKE_OSX_SYSROOT})`

just before

```
execute_process(
         COMMAND "${CMAKE_SOURCE_DIR}/configure"
```

**Comment by venabled** [comment]
Came here from #5661 via @gourdi, the suggestion from #9036 :

> @gourdi
> 
> You might try to do this and rebuild it.
> 
> `set(ENV{SDKROOT} ${CMAKE_OSX_SYSROOT})`
> 
> just before
> 
> ```
> execute_process(
>          COMMAND "${CMAKE_SOURCE_DIR}/configure"
> ```

Seemed to work on my build (Xcode 11.4.1 on macOS 10.15.4)

**Comment by sam20908** [comment]
@offlinemark I am having the exact same issue. I have to install `glib` because it's a dependency of `gtkmm`. Hopefully this gets fixed soon...

**Comment by NancyLi1013** [comment]
Hi all, this issue should be fixed via PR #13100.

Please update to the latest vcpkg and rebuild this port.
